Sources/PackageModel/ToolchainConfiguration.swift

```diff
@@ -52,15 +68,36 @@ public struct ToolchainConfiguration {
         sdkRoot: AbsolutePath? = nil,
         xctestLocation: AbsolutePath? = nil
     ) {
-        self.swiftCompiler = swiftCompiler
-        self.swiftCompilerFlags = swiftCompilerFlags
-        self.libDir = libDir ?? Self.libDir(forBinDir: swiftCompiler.parentDirectory)
-        self.binDir = binDir
-        self.sdkRoot = sdkRoot
-        self.xctestLocation = xctestLocation
+        self.init(
+            swiftCompilerPath: swiftCompiler,
+            swiftCompilerFlags: swiftCompilerFlags,
+            swiftPMLibrariesLocation: libDir.map { .init(root: $0) },
+            sdkRootPath: sdkRoot,
+            xctestPath: xctestLocation
+        )
     }
+}
+
+extension ToolchainConfiguration {
+    public struct SwiftPMLibrariesLocation {
+        public var manifestAPI: AbsolutePath
+        public var pluginAPI: AbsolutePath
+
+        public init(manifestAPI: AbsolutePath, pluginAPI: AbsolutePath) {
+            self.manifestAPI = manifestAPI
+            self.pluginAPI = pluginAPI
+        }
+
+        public init(root: AbsolutePath) {
+            self.init(
```

Comment:
This is a good convenience, I think.  The idea was that if we add more of these, then the client (an IDE for example) should just be able to have a directory of support files, and then it's mostly SwiftPM's business what should be in that directory.  So for a client to not have to enumerate these subdirectories in the default case is good.